### DISCUSSION TITLE
How to work with Record object returned from $app.dao

### DISCUSSION THREAD

**Discussion by moritz4** [pocketbase/pocketbase#5834] [Q&A] [answered]
Hi, I'm having problems working with Records returned from `$app.dao`. I understand I can access the fields using `.get("fieldname")` but it looks like the Records object is an array of objects, why can't I just access it using `.fieldname`?

main.pb.ts
```
nRecordBeforeAuthWithOAuth2Request((e) => {
	// Get the entire 'organisations' collection
	const organisations = $app.dao().findRecordsByExpr("organisations");

	// This works (shows the entire collection )
	console.log(JSON.stringify(organisations));

	// This works
	console.log(organisations[0]?.get("name"));

	// This doesn't print anything. Why not?
	console.log(organisations[0].name);
});
```

**Comment by ganigeorgiev** [comment, marked as answer]
The JSVM return results are often direct mapping/translations of the same Go equivalent APIs.

While it may serialize like an object, the returned `Record` model is NOT a regular JavaScript object. It is the [`Record` Go struct](https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.23/models#Record) and currently there is no support for accessing dynamic fields just as props in the JSVM (_the same apply if you operate with a custom Go `map` type that has methods attached to it_).

It exists somewhere in my todo to experiment with defining a global [`Proxy`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Proxy) but it remains a low priority for now (_it may get supported also if https://github.com/dop251/goja/issues/426 gets eventually implemented_).

**Reply by moritz4** [reply]
Thanks for your prompt reply and clearing that up. 

Any harm if I were to convert the returned Record to an object by running `JSON.parse(JSON.stringify(organisations));`?

**Reply by ganigeorgiev** [reply]
I guess it is fine but note that it may not be very performant and the returned objects cannot be passed to functions and methods that expect a `Record` type as argument.
